src/sparsebase/preprocess/preprocess.h

```diff
@@ -1,760 +1,804 @@
+   */
+  IDType *GetDegrees(format::Format *format,
+                     std::vector<context::Context *> contexts);
+  //! Degree generation implementation function for CSRs
+  /*!
+   *
+   * \param formats A vector containing a single format pointer that should
+   * point at a CSR object \param params a PreprocessParams pointer, though it
+   * is not used in the function \return an array of size
+   * formats[0].get_dimensions()[0] where element i is the degree of the ith
+   * vertex in `formats[0]`
+   */
+  static IDType *GetDegreesCSR(std::vector<format::Format *> formats,
+                               PreprocessParams *params);
+  ~Degrees();
+
+protected:
+  void Register();
+};
+
+//! Find the degree and degree distribution of each vertex in the graph
+//! representation of a format object
+template <typename IDType, typename NNZType, typename ValueType,
+          typename FeatureType>
+class Degrees_DegreeDistribution
+    : public FeaturePreprocessType<
+          std::unordered_map<std::type_index, std::any>> {
+  struct Params : PreprocessParams {};
+
+public:
+  Degrees_DegreeDistribution();
+  Degrees_DegreeDistribution(
+      const Degrees_DegreeDistribution<IDType, NNZType, ValueType, FeatureType>
+          &d);
+  Degrees_DegreeDistribution(std::shared_ptr<Params>);
+  std::unordered_map<std::type_index, std::any>
+  Extract(format::Format *format, std::vector<context::Context *>) override;
+  std::vector<std::type_index> get_sub_ids() override;
+  std::vector<ExtractableType *> get_subs() override;
+  static std::type_index get_feature_id_static();
+
+  //! Degree and degree distribution generation executor function that carries
+  //! out function matching
+  /*!
+   *
+   * \param format a single format pointer to any format
+   * \param contexts vector of contexts that can be used for extracting
+   * features. \return a map with two (type_index, any) pairs. One is a degrees
+   * array of type IDType*, and one is a degree distribution array of type
+   * FeatureType*. Both arrays have the respective metric of the ith vertex in
+   * the ith array element.
+   */
+  std::unordered_map<std::type_index, std::any>
+  Get(format::Format *format, std::vector<context::Context *> contexts);
+
+  //! Degree and degree distribution implementation function for CSRs
+  /*!
+   *
+   * \param format a single format pointer to any format
+   * \param contexts vector of contexts that can be used for extracting
+   * features. \return a map with two (type_index, any) pairs. One is a degrees
+   * array of type IDType*, and one is a degree distribution array of type
+   * FeatureType*. Both arrays have the respective metric of the ith vertex in
+   * the ith array element.
+   */
+  static std::unordered_map<std::type_index, std::any>
+  GetCSR(std::vector<format::Format *> formats, PreprocessParams *params);
+  ~Degrees_DegreeDistribution();
+
+protected:
+  void Register();
+};
+
+enum BitMapSize{

```

Comment:
I was thinking maybe this enum should be moved to the util namespace? Because right now it's reachable like this:

`sparsebase::preprocess:BitMapSize`

And if we're going to be using "bitmaps" often, it might be better to place it in a more general namespace. Thoughts?

Reply:
Sounds good, the way it is reachable at the moment is non-ideal.

Reply:
We actually don't have a file for such enums yet ... I think we should keep it as it is and think of a way to organize enums like this one.